ha/monitor/k8s/monitor.py

```diff
@@ -40,41 +41,45 @@ def __init__(self, wait_time=10):
-
-        self.monitors = []
-
-        # event output in pretty format
-        kwargs = {K8SClientConst.PRETTY : True}
-
-        # Seting a timeout value, 'timout_seconds', for the stream.
-        # timeout value for connection to the server
-        # If do not set then we will not able to stop immediately,
-        # becuase synchronus function watch.stream() will not come back
-        # until catch any event on which it is waiting.
-        kwargs[K8SClientConst.TIMEOUT_SECONDS] = K8SClientConst.VAL_WATCH_TIMEOUT_DEFAULT
-
-        # Get MessageBus producer object for all monitor threads
-        producer = self._get_producer()
-
-        # Change to multiprocessing
-        # Creating NODE monitor object
-        node_monitor = ObjectMonitor(producer, K8SClientConst.NODE, **kwargs)
-        self.monitors.append(node_monitor)
-
-        pod_labels = Conf.get(const.HA_GLOBAL_INDEX, "data_pod_label")
-        pod_label_str = ', '.join(pod_label for pod_label in pod_labels)
-        # TODO : Change 'name' field to 'app' in label_selector if required.
-        kwargs[K8SClientConst.LABEL_SELECTOR] = f'name in ({pod_label_str})'
-
-        # Creating POD monitor object
-        pod_monitor = ObjectMonitor(producer, K8SClientConst.POD, **kwargs)
-        self.monitors.append(pod_monitor)
+        try:
+            # set sigterm handler
+            signal.signal(signal.SIGTERM, self.set_sigterm)
+
+            # Read I/O pod selector label from ha.conf . Will be received from provisioner confstore
+            # provisioner needs to be informed to add it in confstore  (to be added there )
+            ConfigManager.init("k8s_resource_monitor")
+            self._conf_stor_search = ConftStoreSearch()
+
+            self.monitors = []
+
+            # event output in pretty format
+            kwargs = {K8SClientConst.PRETTY : True}
+
+            # Seting a timeout value, 'timout_seconds', for the stream.
+            # timeout value for connection to the server
+            # If do not set then we will not able to stop immediately,
+            # becuase synchronus function watch.stream() will not come back
+            # until catch any event on which it is waiting.
+            kwargs[K8SClientConst.TIMEOUT_SECONDS] = K8SClientConst.VAL_WATCH_TIMEOUT_DEFAULT
+
+            # Get MessageBus producer object for all monitor threads
+            producer = self._get_producer()
+
+            # Change to multiprocessing
+            # Creating NODE monitor object
+            node_monitor = ObjectMonitor(producer, K8SClientConst.NODE, **kwargs)
+            self.monitors.append(node_monitor)
+
+            _, nodes_list = self._conf_stor_search.get_cluster_cardinality()
+            nodes_list = ['27d9b5122785444444444444444']
+            watcher_node_ids = ', '.join(node_id for node_id in nodes_list)
+            kwargs[K8SClientConst.LABEL_SELECTOR] = f'cortx.io/machine-id in ({watcher_node_ids})'
```

Comment:
please confirm that nothing get broken if nodes_list is empty.
[This can happen if cluster.conf was incorrect or did not have any data /server pods. 
In that case we log a warning but populate empty list ]

Reply:
Yes confirmed. Added in the testing details also.